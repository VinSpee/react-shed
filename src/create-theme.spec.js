import keys from 'ramda/src/keys';
import createTheme from './create-theme';

describe('theme creation', () => {
  const THEME = createTheme({
    sizes: 'minor second',
    steps: 30,
    colors: {
      black: '#111111',
      white: '#efefef',
      cyan: 'blue',
      magenta: 'pink',
      yellow: 'merigold',
    },
    fonts: {
      sans: 'System',
      serif: 'Georgia',
      mono: 'Input',
    },
    mqs: [
      'min-width: 600px',
      'min-width: 960px',
      'min-width: 1200px',
    ],
  });

  it('should return a theme object', () => {
    const actual = keys(THEME);
    const expected = ['sizes', 'colors', 'fonts', 'mqs'];
    expect(actual).toEqual(expected);
  });

  it('should return a generated sizes object', () => {
    const actual = THEME.sizes;
    expect(keys(actual)).toHaveLength(121);
    expect(actual).toMatchSnapshot();
  });

  it('should return a generated colors object', () => {
    const actual = THEME.colors;
    expect(keys(actual)).toHaveLength(5);
    expect(actual).toMatchSnapshot();
  });

  it('should return a generated fonts object', () => {
    const actual = THEME.fonts;
    expect(keys(actual)).toHaveLength(3);
    expect(actual).toMatchSnapshot();
  });

  it('should throw if given invalid font format', () => {
    const actual = () => createTheme({
      fonts: ['dude'],
    });
    expect(actual).toThrow(/format/);
  });

  it('should throw if given invalid color format', () => {
    const actual = () => createTheme({
      colors: ['dude'],
    });
    expect(actual).toThrowError(/format/);
  });
});
