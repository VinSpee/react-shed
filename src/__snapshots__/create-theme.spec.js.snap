// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme creation should return a generated colors object 1`] = `
Object {
  "black": "#111111",
  "cyan": "blue",
  "magenta": "pink",
  "white": "#efefef",
  "yellow": "merigold",
}
`;

exports[`theme creation should return a generated fonts object 1`] = `
Object {
  "mono": "Input",
  "sans": "System",
  "serif": "Georgia",
}
`;

exports[`theme creation should return a generated sizes object 1`] = `
Object {
  "z-.1": "-0.9375rem",
  "z-.10": "-0.524460475048727rem",
  "z-.11": "-0.4916816953581815rem",
  "z-.12": "-0.4609515893982952rem",
  "z-.13": "-0.4321421150609018rem",
  "z-.14": "-0.40513323286959535rem",
  "z-.15": "-0.3798124058152457rem",
  "z-.16": "-0.35607413045179287rem",
  "z-.17": "-0.3338194972985558rem",
  "z-.18": "-0.3129557787173961rem",
  "z-.19": "-0.2933960425475588rem",
  "z-.2": "-0.87890625rem",
  "z-.20": "-0.2750587898883364rem",
  "z-.21": "-0.25786761552031534rem",
  "z-.22": "-0.24175088955029567rem",
  "z-.23": "-0.22664145895340218rem",
  "z-.24": "-0.21247636776881454rem",
  "z-.25": "-0.19919659478326365rem",
  "z-.26": "-0.18674680760930965rem",
  "z-.27": "-0.1750751321337278rem",
  "z-.28": "-0.16413293637536983rem",
  "z-.29": "-0.15387462785190922rem",
  "z-.3": "-0.823974609375rem",
  "z-.30": "-0.14425746361116487rem",
  "z-.4": "-0.7724761962890625rem",
  "z-.5": "-0.7241964340209961rem",
  "z-.6": "-0.6789341568946838rem",
  "z-.7": "-0.6365007720887661rem",
  "z-.8": "-0.5967194738332182rem",
  "z-.9": "-0.5594245067186421rem",
  "z-1": "-1rem",
  "z-10": "-1.7875512924265609rem",
  "z-11": "-1.9067213785883317rem",
  "z-12": "-2.033836137160887rem",
  "z-13": "-2.1694252129716127rem",
  "z-14": "-2.3140535605030537rem",
  "z-15": "-2.468323797869924rem",
  "z-16": "-2.6328787177279187rem",
  "z-17": "-2.8084039655764466rem",
  "z-18": "-2.9956308966148764rem",
  "z-19": "-3.195339623055868rem",
  "z-2": "-1.0666666666666667rem",
  "z-20": "-3.408362264592926rem",
  "z-21": "-3.6355864155657875rem",
  "z-22": "-3.8779588432701737rem",
  "z-23": "-4.136489432821518rem",
  "z-24": "-4.41225539500962rem",
  "z-25": "-4.706405754676927rem",
  "z-26": "-5.020166138322056rem",
  "z-27": "-5.35484388087686rem",
  "z-28": "-5.711833472935317rem",
  "z-29": "-6.092622371131005rem",
  "z-3": "-1.1377777777777778rem",
  "z-30": "-6.498797195873071rem",
  "z-4": "-1.2136296296296296rem",
  "z-5": "-1.2945382716049383rem",
  "z-6": "-1.3808408230452676rem",
  "z-7": "-1.472896877914952rem",
  "z-8": "-1.571090003109282rem",
  "z-9": "-1.6758293366499009rem",
  "z.1": "0.9375rem",
  "z.10": "0.524460475048727rem",
  "z.11": "0.4916816953581815rem",
  "z.12": "0.4609515893982952rem",
  "z.13": "0.4321421150609018rem",
  "z.14": "0.40513323286959535rem",
  "z.15": "0.3798124058152457rem",
  "z.16": "0.35607413045179287rem",
  "z.17": "0.3338194972985558rem",
  "z.18": "0.3129557787173961rem",
  "z.19": "0.2933960425475588rem",
  "z.2": "0.87890625rem",
  "z.20": "0.2750587898883364rem",
  "z.21": "0.25786761552031534rem",
  "z.22": "0.24175088955029567rem",
  "z.23": "0.22664145895340218rem",
  "z.24": "0.21247636776881454rem",
  "z.25": "0.19919659478326365rem",
  "z.26": "0.18674680760930965rem",
  "z.27": "0.1750751321337278rem",
  "z.28": "0.16413293637536983rem",
  "z.29": "0.15387462785190922rem",
  "z.3": "0.823974609375rem",
  "z.30": "0.14425746361116487rem",
  "z.4": "0.7724761962890625rem",
  "z.5": "0.7241964340209961rem",
  "z.6": "0.6789341568946838rem",
  "z.7": "0.6365007720887661rem",
  "z.8": "0.5967194738332182rem",
  "z.9": "0.5594245067186421rem",
  "z0": 0,
  "z1": "1rem",
  "z10": "1.7875512924265609rem",
  "z11": "1.9067213785883317rem",
  "z12": "2.033836137160887rem",
  "z13": "2.1694252129716127rem",
  "z14": "2.3140535605030537rem",
  "z15": "2.468323797869924rem",
  "z16": "2.6328787177279187rem",
  "z17": "2.8084039655764466rem",
  "z18": "2.9956308966148764rem",
  "z19": "3.195339623055868rem",
  "z2": "1.0666666666666667rem",
  "z20": "3.408362264592926rem",
  "z21": "3.6355864155657875rem",
  "z22": "3.8779588432701737rem",
  "z23": "4.136489432821518rem",
  "z24": "4.41225539500962rem",
  "z25": "4.706405754676927rem",
  "z26": "5.020166138322056rem",
  "z27": "5.35484388087686rem",
  "z28": "5.711833472935317rem",
  "z29": "6.092622371131005rem",
  "z3": "1.1377777777777778rem",
  "z30": "6.498797195873071rem",
  "z4": "1.2136296296296296rem",
  "z5": "1.2945382716049383rem",
  "z6": "1.3808408230452676rem",
  "z7": "1.472896877914952rem",
  "z8": "1.571090003109282rem",
  "z9": "1.6758293366499009rem",
}
`;
